File: BitManipulation/hamming_distance.py
class Solution(object):
    def hammingDistance(self, x, y):
        """
        :type x: int
        :type y: int
        :rtype: int
        """
        # a bitwise XOR will yield a binary number
        # where there is a 1 in each binary digit where x and y differ
        
        # convert the result to a string using bin(), and count the number of 1's
        return bin(x ^ y).count('1')

File: BitManipulation/HammingDistance.java
class Solution {
    
    public int hammingDistance(int x, int y) {
        
        int xor = x ^ y;
        int counter = 0;
        
        // integers are 32 bits
        for (int i = 0; i < 32; i++) {
            counter += (xor >> i) & 1;
        }
        
        return counter;
        
    }
}
